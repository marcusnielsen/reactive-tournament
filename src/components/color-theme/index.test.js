import ColorTheme, {defaultPrimaryMainHex} from '../color-theme'

test('create default state', () => {
  const colorTheme = ColorTheme({initialPrimaryMainHex: defaultPrimaryMainHex})

  return colorTheme.state_.take(1).forEach((actual) => {
    expect(actual).toMatchSnapshot()
  })
})

test('change color', () => {
  const colorTheme = ColorTheme({initialPrimaryMainHex: defaultPrimaryMainHex})

  const promise = colorTheme.state_.skip(1).take(1).forEach((actual) => {
    expect(actual).toMatchSnapshot()
  })

  colorTheme.actions.triggers.setPrimaryMainHex('#882D60')

  return promise
})
