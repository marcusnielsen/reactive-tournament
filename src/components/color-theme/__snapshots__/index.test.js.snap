exports[`test change color 1`] = `
Object {
  "complementaryMain": "hsl(146.39999999999998, 50.3%, 35.5%)",
  "complementaryMainLight": "hsl(146.39999999999998, 50.3%, 53.2%)",
  "complementarySecondary": "hsl(206.39999999999998, 50.3%, 35.5%)",
  "complementarySecondaryLight": "hsl(206.39999999999998, 50.3%, 53.2%)",
  "primaryMain": "hsl(326.4, 50.3%, 35.5%)",
  "primaryMainLight": "hsl(326.4, 50.3%, 53.2%)",
  "primarySecondary": "hsl(26.399999999999977, 50.3%, 35.5%)",
  "primarySecondaryLight": "hsl(26.399999999999977, 50.3%, 53.2%)",
}
`;

exports[`test create default state 1`] = `
Object {
  "complementaryMain": "hsl(235.79999999999995, 49.8%, 44.5%)",
  "complementaryMainLight": "hsl(235.79999999999995, 49.8%, 66.8%)",
  "complementarySecondary": "hsl(295.79999999999995, 49.8%, 44.5%)",
  "complementarySecondaryLight": "hsl(295.79999999999995, 49.8%, 66.8%)",
  "primaryMain": "hsl(55.80000000000001, 49.8%, 44.5%)",
  "primaryMainLight": "hsl(55.80000000000001, 49.8%, 66.8%)",
  "primarySecondary": "hsl(115.80000000000001, 49.8%, 44.5%)",
  "primarySecondaryLight": "hsl(115.80000000000001, 49.8%, 66.8%)",
}
`;
